src/xhydro/pmp.py

```diff
@@ -0,0 +1,667 @@
+    zg1 = zg_corr.diff("plev", label="upper")
+    zg2 = zg_corr.diff("plev", label="lower")
+    zg_moy = xr.concat([zg1, zg2], dim="variable").sum("variable") / 2
+
+    # Add the pressure layer below the lowest pressure level when the surface altitude is bewlow it (e.g., at  sea level).
+    if add_pre_lay:
+        plev_first = float(zg.plev.max())
+        zg_moy = zg_moy.where(
+            ~((zg_moy == zg_moy.sel(plev=plev_first)) & (~da.isnull())),
+            other=(zg + zg_moy).sel(plev=plev_first),
+        )
+
+    # Betas computation for zones equal to zero.
+    if beta_func:
+        db = da.where(da >= 0, 0)
+        beta = db.where(db <= 0, 1)
+        zg_moy = zg_moy * beta
+
+    # Precipitable water.
+    pw = (zg_moy * hus).sum("plev")
+    pw.name = "precipitable_water"
+    pw.attrs = {
+        "long_name": "Precipitable water",
+        "description": "Precipitable water computed from the specific humidity and geopotential height.",
+        "units": "m",
+    }
+    pw = xclim.core.units.convert_units_to(pw, "mm")
+
+    # Compute the precipitable water for the given window.
+    out = xr.concat(
+        [
+            pw.rolling({"time": window + 1}, center=False)
+            .max(keep_attrs=True)
+            .assign_coords({"window": window})
+            for window in windows
+        ],
+        dim="window",
+    )
+    out["window"].attrs = {
+        "description": "Duration of the event, including antecedent conditions (window + 1)."
+    }
+
+    return out
+
+
+def precipitable_water_100y(
+    pw: xr.DataArray, dist: str, method: str, mf: float = 0.2, rebuild_time: bool = True
+):
+    """Compute the 100-year return period of precipitable water for each month. Based on Clavet-Gaumont et al. (2017).
+
+    Parameters
+    ----------
+    pw : xr.DataArray
+        DataArray containing the precipitable water.
+    dist : str
+        Probability distributions.
+    method : {"ML" or "MLE", "MM", "PWM", "APP"}
+        Fitting method, either maximum likelihood (ML or MLE), method of moments (MM) or approximate method (APP).
+        Can also be the probability weighted moments (PWM), also called L-Moments, if a compatible `dist` object is passed.
+    mf : float
+        Maximum majoration factor of the 100-year event compared to the maximum of the timeseries.
+        Used as an upper limit for the frequency analysis.
+    rebuild_time : bool
+        Whether or not to reconstruct a timeseries with the same time dimensions as `pw`.
+
+    Returns
+    -------
+    xr.DataArray
+        Precipitable water for a 100-year return period.
+
+    Notes
+    -----
+    https://doi.org/10.1016/j.ejrh.2017.07.003
+    """
+    # Compute max monthly and add a «year» dimension.
+    pw_m = pw.resample({"time": "MS"}).max()
+    pw_m = unstack_dates(pw_m, seasons={m: m for m in range(1, 13)}, new_dim="month")
+
+    # Fits distribution
+    # FIXME: Use xhydro.frequency_analysis.local instead
```

Comment:
I think it's better if we do it in another pull request because I still need to work on the wrapper and I'd like people to start testing the PMP module